app/helpers/sessions_helper.rb

```diff
@@ -1,20 +1,54 @@
 module SessionsHelper
+  # Logs in the given user.
   def log_in(user)
     session[:user_id] = user.id
   end
 
+  # Remembers a user in a persistent session.
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  # Returns the current logged-in user (if any).
   def current_user
```

Comment:
there's soo much happening here, please have two methods to handle the different ways of fetching the current user 

eg `find__user__by__session` and `find_user_by_cookies`
implement the aforementioned methods and let them jell into the` if/else` flow